### PR TITLE
[KOGITO-1559] add dependency management for data index storage modules

### PR DESCRIPTION
**Pull request by xiezhang7**
Jira: https://issues.redhat.com/browse/KOGITO-1559

Description: add maven dependency management for new data index modules related to MongoDB storage

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by xiezhang7** [comment]
Assembly:
#487
https://github.com/kiegroup/kogito-apps/pull/205

**Comment by xiezhang7** [comment]
Pull request for data index: https://github.com/kiegroup/kogito-apps/pull/205

File: pom.xml [hunk 1]
```diff
@@ -111,7 +111,7 @@
 
     <version.io.fabric8.kubernetes-client>4.8.0</version.io.fabric8.kubernetes-client>
     <version.io.prometheus>0.5.0</version.io.prometheus>
-    <version.io.quarkus>1.3.1.Final</version.io.quarkus>
+    <version.io.quarkus>1.4.1.Final</version.io.quarkus>
```
Comment (MarianMacik) [inline pom.xml:114]: I think that a quarkus version upgrade should be done separately.

**Comment by xiezhang7** [comment]
Close the RP for now. Will discuss with the community first regarding alternative storage solutions for data index.

**Pull request closed without merging**